e2e/metrics/metrics.go

```diff
@@ -0,0 +1,196 @@
+
+	metrics := []string{
+		"nomad_client_allocated_memory",
+		"nomad_client_host_cpu_user",
+		"nomad_client_host_disk_available",
+		"nomad_client_host_memory_used",
+		"nomad_client_uptime",
+	}
+	// we start with a very long timeout here because it takes a while for
+	// prometheus to be live and for jobs to initially register metrics.
+	timeout := 60 * time.Second
+	require := require.New(f.T())
+
+	for _, metric := range metrics {
+		var results model.Vector
+		var err error
+		require.Eventually(func() bool {
+			results, err = tc.promQuery(metric)
+			if err != nil {
+				return false
+			}
+			instances := make(map[model.LabelValue]struct{})
+			for _, result := range results {
+				instances[result.Metric["instance"]] = struct{}{}
+			}
+			if len(instances) != len(clientNodes) {
+				err = fmt.Errorf("expected metric '%s' for all clients: %v",
+					metric, results)
+				return false
+			}
+			return true
+		}, timeout, 1*time.Second, "prometheus query failed: %v", err)
+
+		// shorten the timeout after the first workload is successfully
+		// queried so that we don't hang the whole test run if something's
+		// wrong with only one of the jobs
+		timeout = 10 * time.Second
+	}
+}
+
+// query promtheus to verify that metrics are being collected
+// from allocations
+func (tc *MetricsTest) queryAllocMetrics(f *framework.F, workloads map[string]string) {
+
+	// we start with a very long timeout here because it takes a while for
+	// prometheus to be live and for jobs to initially register metrics.
+	timeout := 60 * time.Second
+	require := require.New(f.T())
+
+	for jobName, metric := range workloads {
+		query := fmt.Sprintf("%s{exported_job=\"%s\"}", metric, jobName)
+		var results model.Vector
+		var err error
+		require.Eventually(func() bool {
+			results, err = tc.promQuery(query)
+			if err != nil {
+				return false
+			}
+			return true
+		}, timeout, 1*time.Second, "prometheus query failed: %v", err)
+
+		// make sure we didn't just collect a bunch of zero metrics
+		lastResult := results[len(results)-1]
+		require.Greaterf(float64(lastResult.Value), 0.0,
+			"expected non-zero metrics: %v", results,
+		)
+
+		// shorten the timeout after the first workload is successfully
+		// queried so that we don't hang the whole test run if something's
+		// wrong with only one of the jobs
+		timeout = 10 * time.Second
+	}
+}
+
+// TestMetricsWindows runs a collection of jobs that exercise alloc metrics.
+// Then we query prometheus to verify we're collecting client and alloc metrics
+// and correctly presenting them to the prometheus scraper.
+func (tc *MetricsTest) TestMetricsWindows(f *framework.F) {
+	clientNodes, err := e2eutil.ListWindowsClientNodes(tc.Nomad())
+	if err != nil {
+		f.T().Fatalf("could not list client nodes: %v", err)
+	}
+	if len(clientNodes) == 0 {
+		f.T().Skip("no Windows clients")
+	}
+
+	// TODO(tgross): run metrics on Windows, too
```

Comment:
This will be done in #6825

Reply:
lmk if you want a second pair of eyes on windows stuff next week :+1: